src/3p-frame.js

```diff
@@ -327,3 +326,59 @@ export function generateSentinel(parentWindow) {
 export function resetCountForTesting() {
   count = {};
 }
+
+
+/** @const */
+const AMP_MESSAGE_PREFIX = 'amp-';
+
+/** @enum {string} */
+export const MessageType = {
+  // For amp-ad
+  SEND_EMBED_STATE: 'send-embed-state',
+  EMBED_STATE: 'embed-state',
+  SEND_EMBED_CONTEXT: 'send-embed-context',
+  EMBED_CONTEXT: 'embed-context',
+  SEND_INTERSECTIONS: 'send-intersections',
+  INTERSECTION: 'intersection',
+  EMBED_SIZE: 'embed-size',
+  EMBED_SIZE_CHANGED: 'embed-size-changed',
+  EMBED_SIZE_DENIED: 'embed-size-denied',
+
+  // For amp-inabox
+  SEND_POSITIONS: 'send-positions',
+  POSITION: 'position',
+};
+
+/**
+ * Serialize an AMP post message.
+ *
+ * @param type {string}
+ * @param sentinel {string}
+ * @param opt_data {Object=}
+ * @returns {string}
+ */
+export function serializeMessage(type, sentinel, opt_data) {
+  // TODO: consider wrap the data in a "data" field. { type, sentinal, data }
+  return AMP_MESSAGE_PREFIX
+      + JSON.stringify(Object.assign({}, opt_data, {type, sentinel}));
```

Comment:
No need for this object assign anymore? We can use the same `opt_data.type = type, ...`.